src/dbnode/client/session.go

```diff
@@ -2324,6 +2334,16 @@ func (s *session) streamBlocksMetadataFromPeer(
 
 	for moreResults {
 		if err := s.streamBlocksRetrier.Attempt(fetchFn); err != nil {
+			// Check if the error was a hostNotAvailableError. If so, return the
+			// raw (retryable) hostNotAvailableError since the error is technically
+			// retryable but was wrapped to prevent the exponential backoff in the
+			// actual retrier.
+			if isHostNotAvailableError(err) {
```

Comment:
would it be better to get back a bool and the inner error from this function call? can avoid the extra fn call that way and i prefer the encapsulation of the innerNonRetryableError stuff to be hidden behind the hostNotAvailableError utility method